### PR TITLE
[CL Incentives] Implement `setIncentiveRecords` and `getIncentiveRecords`

### PR DESCRIPTION
**Pull request by AlpinYukseloglu**
Closes: #4274

## What is the purpose of the change

This PR implements a more portable version of state reads/writes relating to incentive records with the primary goal of making it simpler to abstract this logic into a separate module in the future. Specifically, it changes incentive records to work based off of pool Ids instead of as a field under pool objects.

## Brief Changelog

- Implement set/get functions for incentive records that do not depend on pool objects
- Remove the `PoolIncentives` field from CL pools

## Testing and Verifying

- The new functions/logic are tested in `incentives_test.go`

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (no)
  - How is the feature or change documented? (not documented)

### CODE REVIEW

**Comment by AlpinYukseloglu** [comment]
Note that this PR was approved in #4283 but preemptively merged. Should merge this directly to main after #4231 is merged.

**Review by czarcas7ic — APPROVED** [review]

**Review by p0mvn — APPROVED** [review]

**Pull request merged**
